processing/src/main/java/io/druid/query/UnionQueryRunner.java

```diff
@@ -41,31 +41,36 @@ public UnionQueryRunner(
   @Override
   public Sequence<T> run(final Query<T> query, final Map<String, Object> responseContext)
   {
-    DataSource dataSource = query.getDataSource();
-    if (dataSource instanceof UnionDataSource) {
+    if (query instanceof BaseQuery) {
```

Comment:
The processing part of multi data source queries is not considered in this patch and will be in a follow-up pr. This method should be fixed to support multi data source queries. I changed to throw an exception if the query is not BaseQuery.